pkg/controller/intrusiondetection/intrusiondetection_controller_test.go

```diff
@@ -274,6 +277,39 @@ var _ = Describe("IntrusionDetection controller tests", func() {
 		})
 	})
 
+	Context("secret availability", func() {
+		BeforeEach(func() {
+			mockStatus.On("SetDegraded", mock.Anything, mock.Anything).Return()
+		})
+
+		It("should not wait on tigera-ee-installer-elasticsearch-access secret when cluster is managed", func() {
+			Expect(c.Create(ctx, &operatorv1.ManagementClusterConnection{
+				ObjectMeta: metav1.ObjectMeta{Name: "tigera-secure"},
+				Spec: operatorv1.ManagementClusterConnectionSpec{
+					ManagementClusterAddr: "127.0.0.1:12345",
+				},
+			})).ToNot(HaveOccurred())
+
+			_, err := r.Reconcile(ctx, reconcile.Request{})
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(mockStatus.AssertNumberOfCalls(nil, "SetDegraded", 0)).To(BeTrue())
+		})
+
+		It("should wait on tigera-ee-installer-elasticsearch-access secret when in a management cluster", func() {
+			Expect(c.Create(ctx, &operatorv1.ManagementCluster{
+				ObjectMeta: metav1.ObjectMeta{Name: "tigera-secure"},
+				Spec: operatorv1.ManagementClusterSpec{
+					Address: "127.0.0.1:12345",
+				},
+			})).ToNot(HaveOccurred())
+
+			_, err := r.Reconcile(ctx, reconcile.Request{})
+			Expect(err).ShouldNot(HaveOccurred())
+			// The missing secret should force utils.ElasticSearch to return a NotFound error which triggers r.status.SetDegraded.
+			Expect(mockStatus.AssertNumberOfCalls(nil, "SetDegraded", 1)).To(BeTrue())
```

Comment:
Instead of doing this it would probably be better to add an "AfterEach" in the top level context and use the AssertExpectations. That means that if you set "On" for a mock and that function wasn't called it would error out. I've been doing this for other projects, I think this was one of the first projects I started mocking with so I think I just forgot.